packages/google_maps_flutter/lib/src/controller.dart

```diff
@@ -119,120 +102,47 @@ class GoogleMapController extends ChangeNotifier {
     notifyListeners();
   }
 
-  /// Starts an animated change of the map camera position.
-  ///
-  /// The returned [Future] completes after the change has been started on the
-  /// platform side.
-  Future<void> animateCamera(CameraUpdate cameraUpdate) async {
-    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-    // https://github.com/flutter/flutter/issues/26431
-    // ignore: strong_mode_implicit_dynamic_method
-    await _channel.invokeMethod('camera#animate', <String, dynamic>{
-      'cameraUpdate': cameraUpdate._toJson(),
-    });
-  }
-
-  /// Changes the map camera position.
+  /// Updates marker configuration.
   ///
-  /// The returned [Future] completes after the change has been made on the
-  /// platform side.
-  Future<void> moveCamera(CameraUpdate cameraUpdate) async {
-    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-    // https://github.com/flutter/flutter/issues/26431
-    // ignore: strong_mode_implicit_dynamic_method
-    await _channel.invokeMethod('camera#move', <String, dynamic>{
-      'cameraUpdate': cameraUpdate._toJson(),
-    });
-  }
-
-  /// Adds a marker to the map, configured using the specified custom [options].
-  ///
-  /// Change listeners are notified once the marker has been added on the
+  /// Change listeners are notified once the update has been made on the
   /// platform side.
   ///
-  /// The returned [Future] completes with the added marker once listeners have
-  /// been notified.
-  Future<Marker> addMarker(MarkerOptions options) async {
-    final MarkerOptions effectiveOptions =
-        MarkerOptions.defaultOptions.copyWith(options);
+  /// The returned [Future] completes after listeners have been notified.
+  Future<void> _updateMarkers(MarkerUpdates markerUpdates) async {
+    assert(markerUpdates != null);
     // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
     // https://github.com/flutter/flutter/issues/26431
     // ignore: strong_mode_implicit_dynamic_method
-    final String markerId = await _channel.invokeMethod(
-      'marker#add',
-      <String, dynamic>{
-        'options': effectiveOptions._toJson(),
-      },
+    await _channel.invokeMethod(
+      'markers#update',
+      markerUpdates._toMap(),
     );
-    final Marker marker = Marker(markerId, effectiveOptions);
-    _markers[markerId] = marker;
     notifyListeners();
```

Comment:
Not related to this PR - We should re-evaluate the single-listener model before we hit API stability.